logger/slog_logger.go

```diff
@@ -0,0 +1,89 @@
+// Copyright © 2018 Banzai Cloud
+// Copyright © 2023 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logger
+
+import (
+	"context"
+	"log/slog"
+	"net"
+	"os"
+	"slices"
+
+	slogmulti "github.com/samber/slog-multi"
+	slogsyslog "github.com/samber/slog-syslog"
+	"github.com/spf13/cast"
+)
+
+var logger *slog.Logger
```

Comment:
Storing a logger in a global variable is an anti-pattern. The best practice is to initialize these kind of services in `main` func and inject them into other services where necessary.